src/Http/Headers/src/ContentDispositionHeaderValue.cs

```diff
@@ -607,10 +608,19 @@ private bool TryDecodeMime(StringSegment input, [NotNullWhen(true)] out string?
 
         // Encode a string using RFC 5987 encoding
         // encoding'lang'PercentEncodedSpecials
+        [SkipLocalsInit]
         private static string Encode5987(StringSegment input)
         {
             var builder = new StringBuilder("UTF-8\'\'");
-            ReadOnlySpan<byte> inputBytes = Encoding.UTF8.GetBytes(input.Value);
+
+            var maxInputBytes = Encoding.UTF8.GetMaxByteCount(input.Length);
+            byte[]? bufferFromPool = null;
+            Span<byte> inputBytes = maxInputBytes <= MaxStackSize
+                ? stackalloc byte[MaxStackSize]
+                : bufferFromPool = ArrayPool<byte>.Shared.Rent(maxInputBytes);
```

Comment:
Hm, now I see in L554 that for the case `> MaxStackSize` just an ordinary byte[] is allocated.
If we are pretty sure that the majority of cases goes to stackalloc, then new byte[] is good "backup".
If there may be frequent hits for `> MaxStackSize` then the array-pool is better suited.

Sorry, that I didn't check this before.

Reply:
No worry, I'll update that part as well.